jina/drivers/__init__.py

```diff
@@ -281,36 +282,8 @@ def docs(self):
             return self.req.docs
 
     def __call__(self, *args, **kwargs):
-        self._traverse_apply(self.docs, *args, **kwargs)
-
-    def _traverse_apply(self, docs: 'DocumentSet', *args, **kwargs) -> None:
-        for path in self._traversal_paths:
-            if path[0] == 'r':
-                self._traverse_rec(docs, None, None, [], *args, **kwargs)
-            for doc in docs:
-                self._traverse_rec(
-                    [doc],
-                    None,
-                    None,
-                    path,
-                    *args,
-                    **kwargs,
-                )
-
-    def _traverse_rec(self, docs, parent_doc, parent_edge_type, path, *args, **kwargs):
-        if path:
-            next_edge = path[0]
-            for doc in docs:
-                if next_edge == 'm':
-                    self._traverse_rec(
-                        doc.matches, doc, 'matches', path[1:], *args, **kwargs
-                    )
-                if next_edge == 'c':
-                    self._traverse_rec(
-                        doc.chunks, doc, 'chunks', path[1:], *args, **kwargs
-                    )
-        else:
-            self._apply_all(docs, parent_doc, parent_edge_type, *args, **kwargs)
+        docset = DocumentSet(self.docs)
+        docset.traverse_apply(self._traversal_paths, self._apply_all, *args, **kwargs)
```

Comment:
```suggestion
        self.docs.traverse_apply(self._traversal_paths, self._apply_all, *args, **kwargs)
```
already in `DocumentSet`, no point to convert it again